protocols/imap/src/main/java/org/apache/james/imap/encode/ExpungeResponseEncoder.java

```diff
@@ -33,7 +34,7 @@
 
     @Override
     public void encode(ExpungeResponse expungeResponse, ImapResponseComposer composer) throws IOException {
-        int messageSequenceNumber = expungeResponse.getMessageSequenceNumber();
+        int messageSequenceNumber = expungeResponse.getMessageSequenceNumber().or(NO_MESSAGE);
```

Comment:
oh ok, I better understand why you need this or

Reply:
I would prefer `NullableMSN` to expost a `public Optional<Integer> asOptionalInt()` method that would allow us to write here:

```
expungeResponse.getMessageSequenceNumber().asOptionalInt().orElse(NO_MESSAGE);
```

This will make the code within NullableMSN more explicit and answers @rouazana  comments?

Reply:
Going from a null-object type to another one before calling `or` looks like additional boilerplate.
It happens that you know Optional null-object well but it doesn't mean we can't implement this pattern ourselves, right?

Reply:
>  it doesn't mean we can't implement this pattern ourselves, right?

Why implement ourselve something that we have at hands and is battle tested.

I, like @rouazana did not understood the `or` method when I read your NullableXX class. Even here I =find the syntax not really explicit.

Reply:
> > it doesn't mean we can't implement this pattern ourselves, right?
> 
> Why implement ourselve something that we have at hands and is battle tested.

Because it's just a `if`. BTW, I don't exactly fear buggy implementation of null-object pattern.
As it's name express, it's a pattern, something you want to apply when you have an common issue.
You are suggesting to DRY and I have to state that I really prefer _ad hoc_ methods and types than generic ones.

> I, like @rouazana did not understood the `or` method when I read your NullableXX class. Even here I =find the syntax not really explicit.

I think that what surprised @rouazana is that he expects `or` of nullableMSN to return an MSN and it actually returns an `int`.

I agree that we lack pattern-matching to handle the null case in a good way. I'll adopt your proposal of using `Optional`.